wgsl/index.bs

```diff
@@ -10156,23 +10192,28 @@ A <dfn noexport>resource</dfn> is an object which provides access to data extern
 and which is not an [=override-declaration=] and not a [[#stage-inputs-outputs|shader stage input or output]].
 Resources are shared by all invocations of the shader.
 
-There are four kinds of resources:
+There are five kinds of resources:
```

Comment:
These edits look inconsistent with Shader Interfaces now. Also the carve outs for immediate data aren't great with the rest of the spec. I wonder if we should instead introduce a new interface for immediates. That might make all this cleaner. Thoughts @dneto0?